keps/sig-api-machinery/3285-subresource-finalizers/README.md

```diff
@@ -0,0 +1,784 @@
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes a way to set different RBACs to `finalizers` field and its entire resource.
+
+## Motivation
+
+`Finalizers` is used by controllers as a pre-deletion hook.
+There is no subresource for `finalizers`, so the only way to grant an actor access to them is to grant access to the whole object.
+On the other hand, some controllers only need to update `finalizers` field and some of its subresources, and don't need to update its entire resource.
+To implement least privilege for such controllers, there should be a way to only allow updating `finalizers`.
+
+### Goals
+
+- Provide a way to set a different RBAC to `finalizers` from its entire resource
+
+### Non-Goals
+
+- Provide a way to set a different RBAC to other than `finalizers` from its entire resource
+- Provide a way to allow updating its entire resource  but disallow updating `finalizers` field
+
+## Proposal
+
+A new endpoint `finalizers` is added as an alternative way to access to `finalizers` field and allow setting a separate RBAC to the endpoint.
+The existing way to acceess `finalizers` field through `metadata.finalizers` remains unchanged.
+
+### User Stories (Optional)
+
+#### Story 1
+
+There is a controller that only updates `status` field and `finalizers` field of a CRD. Administrators of the controller can allow the controller to update only these particular fields.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+- Name collision of the endpoints is needed to be considered.
+- Other existing fields may require the same alternative way to access.
+For example, [candidates](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go) in metadata fields are:
+  - `labels`
+  - `annotations`
+  - `ownerReferences`
+  - `status/conditions`
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+Add `/{resource_name}/finalizers` endpoint to all resources and allow access to `metadata.finalizers` for the resources through the endpoints.
```

Comment:
did we get agreement on the subresource name?  I propose you add an UNRESOLVED block here to capture the ongoing debate, but we don't need to decide to get this KEP merged, IMO.

Reply:
again as a "resource" I would suggest plural noun

Reply:
Added an UNRESOLVED block. Let's keep discussion on this.